pgxn/neon/walproposer.c

```diff
@@ -1024,7 +1024,8 @@ DetermineEpochStartLsn(WalProposer *wp)
 	dth = &wp->safekeeper[wp->donor].voteResponse.termHistory;
 	wp->propTermHistory.n_entries = dth->n_entries + 1;
 	wp->propTermHistory.entries = palloc(sizeof(TermSwitchEntry) * wp->propTermHistory.n_entries);
-	memcpy(wp->propTermHistory.entries, dth->entries, sizeof(TermSwitchEntry) * dth->n_entries);
+	if (dth->n_entries > 0)
+		memcpy(wp->propTermHistory.entries, dth->entries, sizeof(TermSwitchEntry) * dth->n_entries);
```

Comment:
Hm, we then access `wp->propTermHistory.entries` just by index. So in case of `dth->n_entries = 0` we just fill in previously palloc'ed element, which formally seems fine. But I don't know this code, so I wonder, is this `dth->n_entries = 0` case valid at all? Maybe there should be an assert somewhere? cc @arssher

Reply:
Ah, sorry, missed the previous conversation and comment https://github.com/neondatabase/neon/pull/10401#discussion_r1916251891